### PR TITLE
FEAT: UITextView+ViewStyle

### PR DESCRIPTION
**Pull request by matejdecky**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -220,7 +220,7 @@ Styles is available under the MIT license. See the LICENSE file for more info.
 
 ## Authors
 
-|<div align="center"><img src="https://avatars2.githubusercontent.com/u/14109333?s=400&v=4" width="40px;"/><br />[Radim Halfar](https://github.com/radimhalfar)</div>|<div align="center"><img src="https://avatars1.githubusercontent.com/u/560958?s=400&v=4" width="40px;"><br />[Jakub Petrik](https://github.com/jakubpetrik)</div>|
+|<div align="center"><img src="https://avatars2.githubusercontent.com/u/14109333?s=400&v=4" width="40px;"/><br />[Radim Halfar](https://github.com/radimhalfar)</div>|<div align="center"><img src="https://avatars1.githubusercontent.com/u/560958?s=400&v=4" width="40px;"><br />[Jakub Petrik](https://github.com/jakubpetrik)</div>|<div align="center"><img src="https://avatars1.githubusercontent.com/u/25616123?s=400&v=4" width="40px;"><br />[Matěj Děcký](https://github.com/matejdecky)</div>|
 | :---: | :---: |
```
Comment (jakubpetrik) [inline README.md]: fix table layout please

File: Styles/Classes/UIElement+Extensions/UITextView+Styles.m [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    dispatch_once(&onceToken, ^{
+        Class class = [self class];
+        swizzle_instance_method(class, @selector(becomeFirstResponder), @selector(swizzle_becomeFirstResponder));
+        swizzle_instance_method(class, @selector(resignFirstResponder), @selector(swizzle_resignFirstResponder));
+        swizzle_instance_method(class, @selector(awakeFromNib), @selector(swizzle_awakeFromNib));
+    });
+}
+
+- (BOOL)swizzle_becomeFirstResponder {
+    [self updateStylesForState:kEditing];
```
Comment (jakubpetrik) [inline Styles/Classes/UIElement+Extensions/UITextView+Styles.m]: [self swizzle_becomeFirstResponder]

File: Styles/Classes/UIElement+Extensions/UITextView+Styles.m [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    return [super becomeFirstResponder];
+}
+
+- (BOOL)swizzle_resignFirstResponder {
+    [self updateStylesForState:kInactive];
```
Comment (jakubpetrik) [inline Styles/Classes/UIElement+Extensions/UITextView+Styles.m]: [self swizzle_resignFirstResponder]

Don't forget that we are in category here. Not in subclass.

File: Styles/Classes/UIElement+Extensions/UITextView+Styles.m [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    self.textStyles[@(state)] = style;
+    [self applyStyle];
+}
+
+- (void)setLayerStyle:(LayerStyle *)style forState:(UITextViewState)state {
+    self.layerStyles[@(state)] = style;
+    [self applyStyle];
+}
+
+
```
Comment (jakubpetrik) [inline Styles/Classes/UIElement+Extensions/UITextView+Styles.m]: extra line

File: Styles/Classes/UIElement+Extensions/UITextView+Styles.m [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+- (void)updateStylesForState:(UITextViewState)state {
```
Comment (jakubpetrik) [inline Styles/Classes/UIElement+Extensions/UITextView+Styles.m]: rename to applyStyleForState

File: Styles/Classes/UIElement+Extensions/UITextView+Styles.m [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+- (void)textViewChangeEditing:(NSNotification *)notification {
+    if (notification.object != self) {
+        return;
+    }
+    UITextViewState state = notification.name == UITextViewTextDidBeginEditingNotification ? kEditing : kInactive;
+    [super applyStyle];
+    [self updateStylesForState:state];
+}
+
+- (void)updateTextViewUsingTextStyle:(TextStyle *)style {
```
Comment (jakubpetrik) [inline Styles/Classes/UIElement+Extensions/UITextView+Styles.m]: rename to apply...

File: Styles/Classes/UIElement+Extensions/UITextView+Styles.m [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+    NSString *text = self.text;
+    if (!text || !style) {
+        return;
+    }
+    [self setAttributedText:[style applyTo:text]];
+}
+
+- (void)updateTextViewUsingLayerStyle:(LayerStyle *)style {
```
Comment (jakubpetrik) [inline Styles/Classes/UIElement+Extensions/UITextView+Styles.m]: rename to apply...

File: Styles/Classes/UIElement+Extensions/UITextView+Styles.m [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+
+- (void)setLayerStyles:(NSMutableDictionary *)layerStyles {
+    objc_setAssociatedObject(self, @selector(layerStyles), layerStyles, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
+}
+
+- (void)setTextStyles:(NSMutableDictionary *)textStyles {
+    objc_setAssociatedObject(self, @selector(textStyles), textStyles, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
+}
+
+
```
Comment (jakubpetrik) [inline Styles/Classes/UIElement+Extensions/UITextView+Styles.m]: extra line

File: Example/Pods/Pods.xcodeproj/project.pbxproj [hunk 9]
```diff
@@ -359,6 +366,7 @@
 			buildActionMask = 2147483647;
 			files = (
 				6C5491848003C85CEB424075D9E017E0 /* Pods-Styles_Example-dummy.m in Sources */,
+				CCB6CEB82036E19400C49A5C /* UITextView+Styles.m in Sources */,
```
Comment (jakubpetrik) [inline Example/Pods/Pods.xcodeproj/project.pbxproj]: also bump version

File: Styles/Classes/UIElement+Extensions/UITextView+Styles.h [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+    kEditing,
+    kInactive,
+};
+
+@class TextStyle;
+@class LayerStyle;
+
+@interface UITextView (Styles)
+- (void)setTextStyle:(TextStyle *)style forState:(UITextViewState)state UI_APPEARANCE_SELECTOR;
+- (void)setLayerStyle:(LayerStyle *)style forState:(UITextViewState)state UI_APPEARANCE_SELECTOR;
```
Comment (radimhalfar) [inline Styles/Classes/UIElement+Extensions/UITextView+Styles.h]: right now i guess it's ViewStyle

File: Styles/Classes/UIElement+Extensions/UITextView+Styles.m [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+- (void)swizzle_awakeFromNib {
+    [self swizzle_awakeFromNib];
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(textViewChangeEditing:)
+                                                 name:UITextViewTextDidBeginEditingNotification
+                                               object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(textViewChangeEditing:)
+                                                 name:UITextViewTextDidEndEditingNotification
```
Comment (radimhalfar) [inline Styles/Classes/UIElement+Extensions/UITextView+Styles.m]: textViewEditingDidChange

File: Styles/Classes/UIElement+Extensions/UITextInputState.h [hunk 12]
```diff
@@ -0,0 +1,11 @@
+//
+//  UITextInputState.h
+//  Pods
+//
+//  Created by Matěj Děcký on 20.02.18.
+//
+
+typedef NS_ENUM(NSInteger, UITextInputState) {
```
Comment (jakubpetrik) [inline Styles/Classes/UIElement+Extensions/UITextInputState.h]: `UI` prefix is reserved for `UIKit`. Let's go with `TextInputState` only. 🙏

File: Example/Styles/Sources/AppDelegate.swift [hunk 13]
```diff
@@ -157,6 +157,8 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         UITextField.appearance().setViewStyle(blue, for: .inactive)
         UITextField.appearance().setPlaceholderStyle(greenHeadline, for: .inactive)
         UITextField.appearance().setPlaceholderStyle(magentaFootnote, for: .editing)
+        UITextView.appearance().setViewStyle(red, for: .editing)
+        UITextView.appearance().setViewStyle(blue, for: .inactive)
```
Comment (radimhalfar) [inline Example/Styles/Sources/AppDelegate.swift]: how about text style ? 😛

File: Styles/Classes/UIElement+Extensions/UITextView+Styles.m [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+    self.textStyles[@(state)] = style;
+    [self applyStyle];
+}
+
+- (void)setViewStyle:(ViewStyle *)style forState:(UITextInputState)state {
+    self.viewStyles[@(state)] = style;
+    [self applyStyle];
+}
+
+- (void)updateStylesForState:(UITextInputState)state {
```
Comment (jakubpetrik) [inline Styles/Classes/UIElement+Extensions/UITextView+Styles.m]: applyStylesForState:

File: Styles/Classes/UIElement+Extensions/UITextView+Styles.h [hunk 15]
```diff
@@ -0,0 +1,18 @@
+//
+//  UITextView+Styles.h
+//  Pods-Styles_Example
+//
```
Comment (radimhalfar) [inline Styles/Classes/UIElement+Extensions/UITextView+Styles.h]: update header

File: Styles/Classes/UIElement+Extensions/UITextInputState.h [hunk 16]
```diff
@@ -0,0 +1,11 @@
+//
+//  UITextInputState.h
+//  Pods
```
Comment (radimhalfar) [inline Styles/Classes/UIElement+Extensions/UITextInputState.h]: also here

**Pull request merged**